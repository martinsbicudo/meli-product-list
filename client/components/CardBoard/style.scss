.CardBoard {
  box-shadow: 0 0 .1rem .1rem map-get($colors, "meli-gallery"),
    0 0 1rem map-get($colors, "meli-gallery");
  border-radius: .3rem;
  padding: 2rem;
  transition: .1s box-shadow;

  &:hover {
    box-shadow: 0 0 .1rem .1rem map-get($colors, "meli-gallery"),
      0 0 3rem darken(map-get($colors, "meli-gallery"), 4%);
  }

  &Footer {
    margin: -2rem;
    margin-top: 2rem;
    padding: 2rem;
    border-top: .1px solid map-get($colors, "meli-gallery");
    width: 100%;
  }
}
