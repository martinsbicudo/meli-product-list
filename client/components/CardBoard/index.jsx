import React from 'react';
import { node, string, oneOfType } from 'prop-types';
import classnames from 'classnames';
import style from './style.scss';

const CardBoard = ({
  className,
  children,
  footer,
}) => (
  <div
    className={classnames(
      style.CardBoard,
      className,
    )}
  >
    {children}
    {footer && (
      <div className={style.CardBoardFooter}>
        {footer}
      </div>
    )}
  </div>
);

CardBoard.propTypes = {
  className: string,
  children: oneOfType([string, node]).isRequired,
  footer: oneOfType([string, node]),
};

CardBoard.defaultProps = {
  className: '',
  footer: '',
};

export default CardBoard;
