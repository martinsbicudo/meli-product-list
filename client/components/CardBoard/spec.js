import CardBoard from '.';

describe('Components - <CardBoard />', () => {
  const testMessageChildren = 'TEST_CHILDREN';
  const testMessageFooter = 'TEST_FOOTER';
  const createWrapper = ({
    footer = false,
  } = {}) => render(
    <CardBoard
      {...footer && {
        footer,
      }}
    >
      <span>{testMessageChildren}</span>
    </CardBoard>,
  );

  test('Should finds Children', () => {
    const { getByText } = createWrapper();

    expect(getByText(testMessageChildren)).toBeInTheDocument();
  });

  test('Should finds Footer', () => {
    const { getByText } = createWrapper({
      footer: <span>{testMessageFooter}</span>,
    });

    expect(getByText(testMessageFooter)).toBeInTheDocument();
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
