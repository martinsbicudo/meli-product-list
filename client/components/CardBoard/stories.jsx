import React from 'react';
import { storiesOf } from '@storybook/react';
import CardBoard from '.';

storiesOf('CardBoard', module)
  .add('With Children', () => (
    <CardBoard>
      <span>Children CardBoard</span>
    </CardBoard>
  ))
  .add('With Footer', () => (
    <CardBoard footer={<span>Footer CardBoard</span>}>
      <span>Children CardBoard</span>
    </CardBoard>
  ));
