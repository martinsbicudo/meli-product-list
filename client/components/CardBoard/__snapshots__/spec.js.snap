// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - <CardBoard /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <div
    class="CardBoard"
  >
    <span>
      TEST_CHILDREN
    </span>
  </div>
</DocumentFragment>
`;
